src/common/isc_sync.cpp

```diff
@@ -1460,7 +1474,12 @@ SharedMemoryBase::SharedMemoryBase(const TEXT* filename, ULONG length, IpcObject
 		}
 	}
 
+	// time to count our attach to file
+	const USHORT UPPER_BOUND = MAX_USHORT - 1;
+	if (sh_mem_header->mhb_flags >= UPPER_BOUND)
+		(Arg::Gds(isc_random) << "Too many connects to single shared memory file").raise();
```

Comment:
Taking into account that mhb_flags is USHORT and incremented by 2, I'm not sure this code could prevent overflow.
Last valid value is 65534 or MAX_USHORT - 2, for this value condition `mhb_flags >= UPPER_BOUND` is `false` and at next line we will assign 0 to the mhb_flags. Am I wrong ?

Reply:
You are wrong. 
MAX_USHORT is 65535 (0xFFFF), not 65536. I.e. MAX_USHORT - 1 is 65534 and I see no problems this place.

Reply:
Yes, indeed.